src/pages/academy/adminPanel/subcomponents/AddUserPanel.tsx

```diff
@@ -0,0 +1,301 @@
+import {
+  Button,
+  Callout,
+  FileInput,
+  FormGroup,
+  H2,
+  H4,
+  HTMLSelect,
+  Icon,
+  Intent,
+  Position
+} from '@blueprintjs/core';
+import { IconNames } from '@blueprintjs/icons';
+import { Popover2 } from '@blueprintjs/popover2';
+import { GridApi, GridReadyEvent } from 'ag-grid-community';
+import { AgGridReact } from 'ag-grid-react';
+import { uniqBy } from 'lodash';
+import React from 'react';
+import { CSVReader } from 'react-papaparse';
+import { Role } from 'src/commons/application/ApplicationTypes';
+
+import Constants from '../../../../commons/utils/Constants';
+
+export type AddUserPanelProps = OwnProps;
+
+type OwnProps = {
+  handleAddNewUsersToCourse: (users: UsernameRoleGroup[], provider: string) => void;
+};
+
+export type UsernameRoleGroup = {
+  username: string;
+  role: Role;
+  group?: string;
+};
+
+const AddUserPanel: React.FC<AddUserPanelProps> = props => {
+  const [users, setUsers] = React.useState<UsernameRoleGroup[]>([]);
+  const [invalidCsvMsg, setInvalidCsvMsg] = React.useState<string | JSX.Element>('');
+  const gridApi = React.useRef<GridApi>();
+
+  const columnDefs = [
+    {
+      headerName: 'Username',
+      field: 'username'
+    },
+    {
+      headerName: 'Role',
+      field: 'role'
+    },
+    {
+      headerName: 'Group',
+      field: 'group'
+    }
+  ];
+
+  const defaultColumnDefs = {
+    filter: true,
+    resizable: true,
+    sortable: true
+  };
+
+  const onGridReady = (params: GridReadyEvent) => {
+    gridApi.current = params.api;
+  };
+
+  const grid = (
+    <div className="Grid ag-grid-parent ag-theme-balham">
+      <AgGridReact
+        domLayout={'autoHeight'}
+        columnDefs={columnDefs}
+        defaultColDef={defaultColumnDefs}
+        onGridReady={onGridReady}
+        onGridSizeChanged={() => gridApi.current?.sizeColumnsToFit()}
+        rowData={users}
+        rowHeight={36}
+        suppressCellSelection={true}
+        suppressMovableColumns={true}
+        pagination
+      />
+    </div>
+  );
+
+  const htmlSelectOptions = [...Constants.authProviders.entries()].map(([id, _]) => id);
+  const [provider, setProvider] = React.useState(htmlSelectOptions[0]);
+
+  const csvRef = React.useRef<CSVReader>(null);
+
+  const validateCsvInput = (data: any[]) => {
+    const processed: UsernameRoleGroup[] = [...users];
+    let hasInvalidInput = false;
+
+    if (data.length + users.length > 1000) {
+      setInvalidCsvMsg('Please limit each upload to 1000 entries!');
+      hasInvalidInput = true;
+    }
```

Comment:
still hardcoded to 1000 per upload for now. will be relooked when the super user rights are implemented